app/models/memberships/common_api.rb

```diff
@@ -40,7 +40,7 @@ def validate_roles
     # But this method should not save the roles, so we must roll back after checking the validity.
     Role.transaction(requires_new: true) do
       roles_to_destroy, roles_to_update = roles.partition(&:marked_for_destruction?)
-      roles_to_destroy.each { |role| role.delete }
+      roles_to_destroy.each { |role| role.really_destroy! }
```

Comment:
We should call the same as we would in our save method, where we use `really_destroy!` for all roles marked for destruction... 

This still won't destroy the role on validation, because of the transaction